portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -44,16 +44,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     Query query = new Query(ENTITY_TITLE);
     PreparedQuery results = datastore.prepare(query);
     
+    int userChoice = getUserChoice(request);
+    if(userChoice == -1){
+        response.setContentType("text/html");
```

Comment:
You can't return text on error and json on success. Just like a method has only one return type, your servlet should only respond with json or text in every situation but not both. In the case they give an invalid number, just return an empty array. If you follow my next comment, you can get rid of this if statement entirely because -1 will work with that change to return an empty array.

Reply:
Okay, thank you!